### PR TITLE
[SYCLomatic] Support migration of 3 cudaDeviceProp fields.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/test/dpct/device001.cu [hunk 1]
```diff
@@ -160,6 +160,23 @@ int main(int argc, char **argv) {
   if(deviceProp.deviceOverlap){
   //dosomething.
   }
+
+  // CHECK: int Max1d = deviceProp.get_image1d_max();
+  int Max1d = deviceProp.maxTexture1D;
```
Comment (wenhuiNi) [inline clang/test/dpct/device001.cu:165]: please add lit test for this case maxTexture1D be reassign

Comment (tangjj11) [reply clang/test/dpct/device001.cu:165]: fixed

**Review by ziranzha — APPROVED** [review]

File: clang/runtime/dpct-rt/include/dpct/device.hpp [hunk 2]
```diff
@@ -162,6 +162,9 @@ class device_info {
   unsigned int get_global_mem_cache_size() const {
     return _global_mem_cache_size;
   }
+  int get_image1d_max() const { return _image1d_max; }
+  auto get_image2d_max() { return _image2d_max; }
```
Comment (ziranzha) [inline clang/runtime/dpct-rt/include/dpct/device.hpp:167]: Need add a overload method with "const" qualifier?

**Review by ziranzha — APPROVED** [review]

File: clang/runtime/dpct-rt/include/dpct/device.hpp (hunk 2 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/device.hpp:167]: fixed

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**